CONTRIBUTING.md

```diff
@@ -45,22 +45,119 @@ To add a new plugin you need to do the following.
   -> See the [template](plugins/TEMPLATE.nix).
 
-3. Add the necessary parameters for the `mkNeovimPlugin`/`mkVimPlugin`:
-  - `name`: The name of the plugin. The resulting nixvim module will have `plugins.<name>` as a path.\
-    For a plugin named `foo-bar.nvim`, set this to `foo-bar` (subject to exceptions).
-  - `originalName`: The "real" name of the plugin (i.e. `foo-bar.nvim`). This is used mostly in documentation.
-  - `package`: The nixpkgs package attr for this plugin
-     e.g. `"foo-bar-nvim` for `pkgs.vimPlugins.foo-bar-nvim`, or `[ "hello" "world" ]` for `pkgs.hello.world`.
-  - `maintainers`: Register yourself as a maintainer for this plugin:
-    - e.g. `maintainers = [ lib.maintainers.JosephFourier ]`
-    - If you are already registered as a [`nixpkgs` maintainer][nixpkgs-maintainers], there is nothing more to do.
-    - Otherwise, you should add yourself to our [`maintainers.nix`](lib/maintainers.nix) list.
-    - See the documentation at the top of the [`nixpkgs` maintainers list][nixpkgs-maintainers] for more detail.
-  - `settingsOptions`: All or some (only the most important ones) option declarations for this plugin settings.\
-    See below for more information
-  - `settingsExample`: An example of what could the `settings` attrs look like.
-
-[nixpkgs-maintainers]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix
+3. Add the necessary parameters for the [`mkNeovimPlugin`](#mkneovimplugin)/[`mkVimPlugin`](#mkvimplugin)
+
+#### `mkNeovimPlugin`
+
+The `mkNeovimPlugin` function provide a standardize way to create a `Neovim` plugin.
+
+| Parameter                  | Description                                                                                                                                                                                                 | Required | Default Value |
+|----------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|---------------|
+| **name**                   | The name of the plugin.                                                                                                                                                                                      | Yes      | N/A           |
+| **maintainers**            | Maintainers for the plugin.                                                                                                                                                                                  | Yes      | N/A           |
+| **url**                    | The URL of the plugin's repository.                                                                                                                                                                          | Yes      | If not set, `package`'s' `homepage` attribute |
```

Comment:
```suggestion
| **url**                    | The URL of the plugin's repository.                                                                                                                                                                          | No     | If not set, `package`'s' `homepage` attribute |
```
I've never passed url to a `mkNeovimPlugin`, so I wouldn't say that's required. The default value even explains why it's not required.